File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/SEMINAR/S1/untitled/Sem1-MAP/src/StackContainer.java
import java.util.ArrayList;
import java.util.List;

public class StackContainer implements Container{

    private List<Task> tasks=new ArrayList<>();

    @Override
    public Task remove() {
        if(tasks.isEmpty())
            return null;
        return tasks.remove(tasks.size()-1);
    }

    @Override
    public void add(Task task) {
        tasks.add(task);
    }

    @Override
    public int size() {
        return tasks.size();
    }

    @Override
    public boolean isEmpty() {
        return tasks.isEmpty();
    }
}

File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/LABORATOR/lab 11 - homework/lab11/lab11/container/StackContainer.cs
namespace lab11;

public class StackContainer : Container
{
    private Task[] tasks;
    private int top;

    public StackContainer(int capacity)
    {
        tasks = new Task[capacity];
        top = -1;

    }
    
    public Task Remove()
    {
        if (top > -1)
        {
            Task task = tasks[top];
            tasks[top] = null;
            top--;
            return task;
        }
        throw new InvalidOperationException("Stack container is empty");
    }

    public void Add(Task task)
    {
        if (top < tasks.Length-1)
        {
            top++;
            tasks[top] = task;
        }
        else throw new InvalidOperationException("Stack container is full");
        
        

    }

    public int Size()
    {
        return top + 1;
    }

    public bool IsEmpty()
    {
        return top == -1;
    }
}